src/test/java/cat/udl/eps/entsoftarch/webingogeiapi/steps/EditGameStepDefs.java

```diff
@@ -0,0 +1,212 @@
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @And("^I edit the pricePerCard to be (\\d+.\\d+) for the game with id (\\d+)$")
+    public void iEditThePricePerCardToBeForTheGameWithId(int newpricePerCard, int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("pricePerCard",newpricePerCard);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                put("/games/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @And("^The pricePerCard has been edited to be (\\d+.\\d+) for the game with id (\\d+)$")
+    public void thePricePerCardHasBeenEditedToBeForTheGameWithId(int pricePerCard, int id) throws Exception {
+        Assert.assertNotNull("Location not null", idc);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get(idc)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+        String game = stepDefs.result.andReturn().getResponse().getContentAsString();
+        String a = (JsonPath.read(game, "$._links.game.href"));
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get(a)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print()).andExpect(jsonPath("$._links.self.href", endsWith("/games/" + String.valueOf(id))));
+
+    }
+
+    @And("^The game with id (\\d+) has been edited$")
+    public void theGameWithIdHasBeenEdited(int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("id",id);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/games/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @Given("^Exists an admin \"([^\"]*)\" with password \"([^\"]*)\"$")
+    public void existsAnAdminWithPassword(String admin, String password) throws Throwable {
+        // Write code here that turns the phrase above into concrete actions
+        JSONObject game = new JSONObject();
+        game.put("admin", admin);
+        game.put("password", password);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/games")
```

Comment:
Complete mess, check if admin exists but really tries to create a game with admin and password attributes!